bridges/centralized-ethereum/src/actors/watch_dog.rs

```diff
@@ -0,0 +1,364 @@
+        let running_secs = self.start_ts.unwrap().elapsed().as_secs();
+
+        let fut = async move {
+            let mut status = "up-and-running".to_string();
+
+            if let Err(err) = check_wit_connection_status(&wit_jsonrpc_socket).await {
+                status = err;
+            }
+            let wit_client = JsonRpcClient::start(&wit_jsonrpc_socket)
+                .expect("cannot start JSON/WIT connection");
+            let wit_account = match fetch_wit_account(&wit_client).await {
+                Ok(pkh) => pkh,
+                Err(err) => {
+                    if status.eq("up-and-running") {
+                        status = err;
+                    }
+                    None
+                }
+            };
+
+            let wit_balance = match wit_account.clone() {
+                Some(pkh) => match fetch_wit_account_balance(&wit_client, pkh.as_str()).await {
+                    Ok(wit_balance) => wit_balance,
+                    Err(err) => {
+                        if status.eq("up-and-running") {
+                            status = err;
+                        }
+                        None
+                    }
+                },
+                None => None,
+            };
+
+            let wit_utxos_above_threshold = match wit_account.clone() {
+                Some(pkh) => {
+                    match fetch_wit_account_count_utxos_above(
+                        &wit_client,
+                        pkh.as_str(),
+                        wit_utxo_min_value_threshold,
+                    )
+                    .await
+                    {
+                        Ok(wit_utxos_above_threshold) => wit_utxos_above_threshold,
+                        Err(err) => {
+                            if status.eq("up-and-running") {
+                                status = err;
+                            }
+                            None
+                        }
+                    }
+                }
+                None => None,
+            };
+
+            let eth_balance = match check_eth_account_balance(&eth_jsonrpc_url, eth_account).await {
+                Ok(Some(eth_balance)) => {
+                    let eth_balance: f64 = eth_balance.to_string().parse().unwrap_or_default();
+                    //Some(Unit::Wei(&eth_balance.to_string()).to_eth_str().unwrap_or_default()),
+                    Some(eth_balance / 1000000000000000000.0)
+                }
+                Ok(None) => None,
+                Err(err) => {
+                    if status.eq("up-and-running") {
+                        status = err;
+                    }
+                    None
+                }
+            };
+
+            let dr_database = DrDatabase::from_registry();
+            let (_, drs_pending, drs_finished, _) =
+                dr_database.send(CountDrsPerState).await.unwrap().unwrap();
+
+            let mut metrics: String = "{".to_string();
```

Comment:
I'd move the logic related with the population of `metrics` to its own function

Reply:
i prefer not to, as tracing logs is the ultimate goal of the watch_global_status function itself. Besides, that function would only be called from single point, and it would require passing all metrics by parameter.